### PR TITLE
Fix handling of queued ajax notification

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Should be reviewed commit by commit.

### CODE REVIEW

**Comment by trasher** [comment]
Can [Rename notification 'sendmail' action to 'send'](https://github.com/glpi-project/glpi/pull/10679/commits/494bdbb73eeaa3b3bda295b3035ea3aade887e74) impact plugins?

**Comment by cedric-anne** [comment]
> Can [Rename notification 'sendmail' action to 'send'](https://github.com/glpi-project/glpi/pull/10679/commits/494bdbb73eeaa3b3bda295b3035ea3aade887e74) impact plugins?

A plugin that has a class that overrides `QueuedNotification` may be impacted, but I do not see how a plugin could override the whole notification queue system, so I guess such a plugin does not exists.

File: src/CommonDBTM.php [hunk 1]
```diff
@@ -5955,7 +5955,7 @@ public function getMassiveActionsForItem(): MassiveAction
             $params['entity_restrict'] = $this->getEntityID();
         }
 
-        return new MassiveAction($params, [], 'initial', true);
+        return new MassiveAction($params, [], 'initial', $this->fields['id']);
```
Comment (orthagh) [inline src/CommonDBTM.php:5958]:
```suggestion
        return new MassiveAction($params, [], 'initial', ($this->fields['id'] > 0));
```

we're waiting for a boolean, maybe let the parameter clear, no ?

**Review by orthagh — APPROVED** [review]
globally ok with the pr

File: src/CommonDBTM.php (hunk 1 above)
Comment (cedric-anne) [reply src/CommonDBTM.php:5958]: No, an `id` has to be used to be able to filter single actions, see `MassiveAction::getAllMassiveActions()`.
I will fix PHPDoc and expected types to make this clear.

Comment (orthagh) [reply src/CommonDBTM.php:5958]: ok

Comment (cedric-anne) [reply src/CommonDBTM.php]: See last commit. I found another bad param usage in API, so I guess changing the signature is usefull.

**Review by trasher — APPROVED** [review]

**Pull request merged**